src/Controller/ModelController.php

```diff
@@ -131,4 +133,78 @@ public function view($id) : ?Response
 
         return null;
     }
+
+    /**
+     * save property types (add/edit/delete)
+     *
+     * @return {void}
+     */
+    public function savePropertiesJson()
+    {
+        $payload = $this->request->getData();
+
+        $this->request->allowMethod(['post']);
+        $response = [];
+
+        try {
+            if (!empty($payload)) {
+                extract($payload);
```

Comment:
I don`t really like `extract()` - code is not so clear and readable, you don't know where variables come from...  we never use it actually

Reply:
also having 3 methods on add / edit / remove operations will make `savePropertiesJson` method more readable

Reply:
ok, i'll refactor it if it's bad practice